src/Shared/Components/CICDHistory/service.tsx

```diff
@@ -42,21 +42,26 @@ const getTriggerDetailsQuery = (fetchIdData) => {
     return ''
 }
 
-export function getTriggerDetails({
+export async function getTriggerDetails({
     appId,
     envId,
     pipelineId,
     triggerId,
     fetchIdData,
 }): Promise<TriggerDetailsResponseType> {
-    if (triggerId) {
-        return get(
-            `${ROUTES.APP}/cd-pipeline/workflow/trigger-info/${appId}/${envId}/${pipelineId}/${triggerId}${getTriggerDetailsQuery(fetchIdData)}`,
-        )
-    }
-    return get(
-        `${ROUTES.APP}/cd-pipeline/workflow/trigger-info/${appId}/${envId}/${pipelineId}/last${getTriggerDetailsQuery(fetchIdData)}`,
+    const response = await get<TriggerDetailsResponseType['result']>(
+        triggerId
+            ? `${ROUTES.APP}/cd-pipeline/workflow/trigger-info/${appId}/${envId}/${pipelineId}/${triggerId}${getTriggerDetailsQuery(fetchIdData)}`
```

Comment:
Suggestion: `${ROUTES.APP}/cd-pipeline/workflow/trigger-info/${appId}/${envId}/${pipelineId} is constant and can we create route